gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -464,3 +469,179 @@ def _get_variant_calls(self, record, formats):
         call.info[field] = data
       calls.append(call)
     return calls
+
+
+class NucleusParser(VcfParser):
+  """An Iterator for processing a single VCF file using Nucleus."""
+
+  def __init__(self,
+               file_name,  # type: str
+               range_tracker,  # type: range_trackers.OffsetRangeTracker
+               file_pattern,  # type: str
+               compression_type,  # type: str
+               allow_malformed_records,  # type: bool
+               representative_header_lines=None,  # type:  List[str]
+               **kwargs  # type: **str
+              ):
+    # type: (...) -> None
+    super(NucleusParser, self).__init__(file_name,
+                                        range_tracker,
+                                        file_pattern,
+                                        compression_type,
+                                        allow_malformed_records,
+                                        representative_header_lines,
+                                        **kwargs)
+    # This member will be properly initiated in _init_with_header().
+    self._vcf_reader = None
+    # These members will be properly initiated in _extract_header_fields().
+    self._is_info_repeated = {}
+    self._is_format_repeated = {}
+    # This is a temporary solution until from_string will be fixed.
+    self._temp_local_file = self._write_to_local_file(file_name)
+
+  def _write_to_local_file(self, remote_file_name):
+    (temp_file, temp_file_name) = tempfile.mkstemp(text=True)
+    with FileSystems.open(remote_file_name) as f:
+      while True:
+        line = f.readline()
+        if line:
+          os.write(temp_file, line)
+        else:
+          os.close(temp_file)
+          break
+    return temp_file_name
+
+  def _store_to_temp_local_file(self, header_lines):
+    (temp_file, temp_file_name) = tempfile.mkstemp(text=True)
+    for line in header_lines:
+      os.write(temp_file, line)
+    os.close(temp_file)
+    return temp_file_name
+
+  def _init_with_header(self, header_lines):
+    # This header line is needed by Nucleus.
+    header_lines = ['##fileformat=VCFv4.2'] + header_lines
```

Comment:
Please make this a constant.

Reply:
What do you think of this suggestion: define `FILE_FORMAT_HEADER_TEMPLATE='##fileformat=VCF{VERSION}'`.
You can then do `startswith(FILE_FORMAT_HEADER_TEMPLATE.format(VERSION='')` ...if it doesn't match, you can add `FILE_FORMAT_HEADER_TEMPLATE.format('v4.0')`. I think it's ok to hard-code the 'v.4.0' for simplicity.